lib/app_profiler/profiler.rb

```diff
@@ -1,86 +1,34 @@
 # frozen_string_literal: true
 
-require "stackprof"
-
 module AppProfiler
   module Profiler
-    DEFAULTS = {
-      mode: :cpu,
-      raw: true,
-    }.freeze
-
     class << self
-      def run(params = {})
-        started = start(params)
-
-        yield
-
-        return unless started
-
-        stop
-        results
-      ensure
-        # Only stop the profiler if profiling was started in this context.
-        stop if started
+      def run(params = {}, &block)
+        AppProfiler.profiler_backend.run(params, &block)
```

Comment:
I assume there has got to be a better ruby-ish way of doing this than just wrapping the call, but I couldn't pick between delegates / forwardable, or trying to do some sort of mix-in thing. For now I figured i'd just do the simplest thing and see what comes out of the review process.

Reply:
I think we can just get rid of this class/file entirely. 

We will need to change methods in `AppProfiler`.

```
   def run(*args, &block)
      Profiler.run(*args, &block)
   end
```

to 

```
   def run(*args, &block)
      profiler_backend.run(*args, &block)
   end
```